client/attest.go

```diff
@@ -109,6 +115,30 @@ func (k *Key) getCertificateChain(client *http.Client) ([][]byte, error) {
 	return nil, fmt.Errorf("max certificate chain length (%v) exceeded", maxCertChainLength)
 }
 
+// Does best effort to get the SEV-SNP hardware rooted attestation, but won't fail fatally
+func getSevSnpAttestationReport(attestation *pb.Attestation, opts AttestOpts) {
+	var err error
+	device := opts.SevSnpDevice
+	if device == nil {
+		device, err = sg.OpenDevice()
+		if err != nil { // No access or not an SEV-SNP guest.
+			return
+		}
+	}
+	var snpNonce [64]byte
```

Comment:
TPM nonces are limited to the max size of the TPM2B_Name. So this is ~34 bytes since all TPM2s need to support SHA256. This *can* be bigger, but not necessarily. If this size doesn't match SNP expectations/requirements, we should consider getting a nonce from another source

Reply:
Done.